contracts/libraries/VaultLifecycleTreasuryBare.sol

```diff
@@ -0,0 +1,827 @@
+        address _keeper;
+        address _feeRecipient;
+        uint256 _managementFee;
+        uint256 _performanceFee;
+        string _tokenName;
+        string _tokenSymbol;
+        address _optionsPremiumPricer;
+        address _strikeSelection;
+        uint32 _premiumDiscount;
+        uint256 _auctionDuration;
+        uint256 _period;
+        uint256 _maxDepositors;
+        uint256 _minDeposit;
+    }
+
+    /**
+     * @notice Sets the next option the vault will be shorting, and calculates its premium for the auction
+     * @param strikeSelection is the address of the contract with strike selection logic
+     * @param optionsPremiumPricer is the address of the contract with the
+       black-scholes premium calculation logic
+     * @param premiumDiscount is the vault's discount applied to the premium
+     * @param closeParams is the struct with details on previous option and strike selection details
+     * @param vaultParams is the struct with vault general data
+     * @param vaultState is the struct with vault accounting state
+     * @return otokenAddress is the address of the new option
+     * @return premium is the premium of the new option
+     * @return strikePrice is the strike price of the new option
+     * @return delta is the delta of the new option
+     */
+    function commitAndClose(
+        address strikeSelection,
+        address optionsPremiumPricer,
+        uint256 premiumDiscount,
+        CloseParams calldata closeParams,
+        Vault.VaultParams storage vaultParams,
+        Vault.VaultState storage vaultState
+    )
+        external
+        returns (
+            address otokenAddress,
+            uint256 premium,
+            uint256 strikePrice,
+            uint256 delta
+        )
+    {
+        uint256 expiry;
+
+        // uninitialized state
+        if (closeParams.currentOption == address(0)) {
+            expiry = getNextExpiry(block.timestamp, closeParams.period);
+        } else {
+            expiry = getNextExpiry(
+                IOtoken(closeParams.currentOption).expiryTimestamp(),
+                closeParams.period
+            );
+        }
+
+        bool isPut = vaultParams.isPut;
+        address underlying = vaultParams.underlying;
+        address asset = vaultParams.asset;
+
+        (strikePrice, delta) = closeParams.lastStrikeOverrideRound ==
+            vaultState.round
+            ? (closeParams.overriddenStrikePrice, 0)
+            : IStrikeSelection(strikeSelection).getStrikePrice(expiry, isPut);
+
+        require(strikePrice != 0, "!strikePrice");
+
+        // retrieve address if option already exists, or deploy it
+        otokenAddress = getOrDeployOtoken(
+            closeParams,
+            vaultParams,
+            underlying,
+            asset,
+            strikePrice,
+            expiry,
+            isPut
+        );
+
+        // get the black scholes premium of the option
+        premium = optionsPremiumPricer == address(0)
```

Comment:
can we acctually do `== address(1)`

Reply:
Addressed in [e34465b](https://github.com/ribbon-finance/ribbon-v2/pull/293/commits/e34465b30162c78b1c9313d3468d42cd1c325bc4)